back/src/bsffs/compat.ts

```diff
@@ -106,11 +108,22 @@ export function toBsffDestination(
  * Compute BSFF status based on the acceptation and processing information
  * of each packaging
  */
-export async function getStatus(bsff: Bsff & { packagings: BsffPackaging[] }) {
+export async function getStatus(
```

Comment:
Ça date d'avant cette PR, mais je trouve cette fonction bien complexe et exigeante à suivre, notamment le reduce.

Reply:
J'ai des vieux réflexes de programmation fonctionnelle qui m'interdise de muter des variables... J'ai réécrit avec des variables et une boucle `for`, tu me diras si tu trouves ça plus clair. Dans tous les cas, ça reste un peu galère de calculer le statut du bordereau à partir de l'état de chaque contenant.

Reply:
C'est très subjectif mais c'est pas le reduce qui me posait pb, plutôt l'enchaînement de ternary où il est très facile d'introduire une régression avec un ! en plus ou en moins par exemple

Reply:
Ouais ce qui introduit de la complexité supplémentaire c'est d'aller voir ce qui se passe sur le `lastPackaging`. Je pense que ça pourrait être mieux géré en introduisant une colonne pour le traitement final qui serait recalculé en asynchrone dès qu'un traitement a lieu en aval (dans le graphe). Bref je garde en tête d'améliorer la lisibilité de cette fonction.